cli/running/watchdog.go

```diff
@@ -28,13 +30,22 @@ type Watchdog struct {
 	// done channel used to inform the signal handle goroutine
 	// about termination of the Instance.
 	done chan bool
+	// createInstance is a function used for instance creation.
+	createInstance createInstanceFunc
+	// args is arguments passed to the program.
+	args []string
```

Comment:
The watchdog must not know anything about instance arguments or about `cmdCtx`. This is none of the watchdog's deal.
In the `watchdog.go` creat an interface like:
```
type InctanceBuilder interface {
    CreateInstance() *Instance
}
```
Add to `Watchdog struct` `InstBuilder *InctanceBuilder`

Add implementation of interface to `running.go` like:
```
type InstBuilderImpl struct {
    configPath string
    .....
}

func (builder *InstBuilderImpl) CreateInstance() *Instance {
    // builder.configPath ...
}
```

Reply:
Changed